stdlib/public/core/UnsafeBufferPointer.swift.gyb

```diff
@@ -666,24 +676,71 @@ extension Unsafe${Mutable}BufferPointer {
   /// `Element` type.
   ///
   /// - Note: Only use this method to rebind the buffer's memory to a type
-  ///   with the same size and stride as the currently bound `Element` type.
-  ///   To bind a region of memory to a type that is a different size, convert
-  ///   the buffer to a raw buffer and use the `bindMemory(to:)` method.
+  ///   that is layout compatible with the currently bound `Element` type.
+  ///   The stride of the temporary type (`T`) may be an integer multiple
+  ///   or a whole fraction of `Element`'s stride.
+  ///   To bind a region of memory to a type that does not match these
+  ///   requirements, convert the buffer to a raw buffer and use the
+  ///   `bindMemory(to:)` method.
+  ///   If `T` and `Element` have different alignments, this buffer's
+  ///   `baseAddress` must be aligned with the larger of the two alignments.
   ///
   /// - Parameters:
   ///   - type: The type to temporarily bind the memory referenced by this
-  ///     buffer. The type `T` must have the same size and be layout compatible
+  ///     buffer. The type `T` must be layout compatible
   ///     with the pointer's `Element` type.
   ///   - body: A closure that takes a ${Mutable.lower()} typed buffer to the
-  ///     same memory as this buffer, only bound to type `T`. The buffer argument 
-  ///     contains the same number of complete instances of `T` as the original  
-  ///     buffer’s `count`. The closure's buffer argument is valid only for the 
-  ///     duration of the closure's execution. If `body` has a return value, that 
-  ///     value is also used as the return value for the `withMemoryRebound(to:_:)` 
+  ///     same memory as this buffer, only bound to type `T`. The buffer
+  ///     parameter contains a number of complete instances of `T` based
+  ///     on the capacity of the original buffer and the stride of `Element`.
+  ///     The closure's buffer argument is valid only for the duration of the
+  ///     closure's execution. If `body` has a return value, that value
+  ///     is also used as the return value for the `withMemoryRebound(to:_:)`
   ///     method.
+  ///   - buffer: The buffer temporarily bound to `T`.
   /// - Returns: The return value, if any, of the `body` closure parameter.
   @inlinable // unsafe-performance
+  @_alwaysEmitIntoClient
+% if Mutable:
+  @_silgen_name("$_swift_UnsafeMutableBufferPointer_se0333_withMemoryRebound")
```

Comment:
These getting-out-of-the-way silgen names are a new thing in the stdlib, as far as I can tell.
Is this naming scheme reasonable? Too fussy?

Reply:
I think that these names are fine; I like that they explicitly point to the proposal for reference.